core/src/main/java/ai/timefold/solver/core/api/score/stream/ConstraintBuilder.java

```diff
@@ -20,12 +21,26 @@ default Constraint asConstraint(String constraintName) {
     /**
      * Builds a {@link Constraint} from the constraint stream.
      * The {@link ConstraintRef#packageName() constraint package} defaults to the package of the {@link PlanningSolution} class.
+     * The constraint will be placed in the {@link Constraint#DEFAULT_CONSTRAINT_GROUP default constraint group}.
      *
      * @param constraintName never null, shows up in {@link ConstraintMatchTotal} during score justification
      * @param constraintDescription never null
      * @return never null
      */
-    Constraint asConstraintDescribed(String constraintName, String constraintDescription);
+    default Constraint asConstraintDescribed(String constraintName, String constraintDescription) {
+        return asConstraintDescribed(constraintName, constraintDescription, Constraint.DEFAULT_CONSTRAINT_GROUP);
+    }
+
+    /**
+     * Builds a {@link Constraint} from the constraint stream.
+     * The {@link ConstraintRef#packageName() constraint package} defaults to the package of the {@link PlanningSolution} class.
+     *
+     * @param constraintName never null, shows up in {@link ConstraintMatchTotal} during score justification
+     * @param constraintDescription never null
+     * @param constraintGroup never null, only allows alphanumeric characters, "-" and "_"
```

Comment:
The regex is using `\p{So}`, which means we will accept other symbols.

Reply:
True. Geoffrey and I thought it'd be nice to allow Emoji, for demo purposes. At the same time, does it need to be documented?

Reply:
Just reading the documentation, I wouldn't assume that we accept these symbols.